### PR TITLE
Update 1.0.4 

### PR DESCRIPTION
**Pull request by vtkalek**
Fix issues,
Add Dot opacity property
Add axis color property
Add new basket "Counter"
Add UT

### CODE REVIEW

File: capabilities.json [hunk 1]
```diff
@@ -17,12 +25,16 @@
         {
           "Date": {
             "min": 0,
-            "max": 1
+            "max": 4
```
Comment (ignatvilesov) [inline capabilities.json]: Why do we allow 4 fields now?

File: capabilities.json [hunk 2]
```diff
@@ -113,6 +137,28 @@
         }
       }
     },
+    "axisoptions": {
```
Comment (ignatvilesov) [inline capabilities.json]: `axisPosition` ? It'd be better to use Camel Case

File: CHANGELOG.md [hunk 3]
```diff
@@ -1,3 +1,8 @@
+## 1.0.4
```
Comment (ignatvilesov) [inline CHANGELOG.md]: It'd be better to increase up to `1.1.0`

File: package.json [hunk 4]
```diff
@@ -1,7 +1,7 @@
 {
   "name": "powerbi-visuals-linedotchart",
   "description": "LineDot Chart",
-  "version": "1.0.3",
+  "version": "1.0.4",
```
Comment (ignatvilesov) [inline package.json]: It'd be better to increase up to `1.1.0` because there're new features

File: pbiviz.json [hunk 5]
```diff
@@ -4,7 +4,7 @@
     "displayName": "LineDot Chart",
     "guid": "LineDotChart1460463831201",
     "visualClassName": "LineDotChart",
-    "version": "1.0.3",
+    "version": "1.0.4",
```
Comment (ignatvilesov) [inline pbiviz.json]: It'd be better to increase up to `1.1.0`

File: src/settings.ts [hunk 6]
```diff
@@ -28,12 +28,19 @@ module powerbi.extensibility.visual {
     import DataViewObjectsParser = powerbi.extensibility.utils.dataview.DataViewObjectsParser;
 
     export class LineDotChartSettings extends DataViewObjectsParser {
+        public isCounterDateTime: boolean = false;
+        public axisoptions: AxisSettings = new AxisSettings();
```
Comment (ignatvilesov) [inline src/settings.ts]: `axisPosition`

File: src/visual.ts [hunk 7]
```diff
@@ -256,7 +256,7 @@ module powerbi.extensibility.visual {
                 || _.isEmpty(categorical.Values[0].values)) {
                 return null;
             }
-
+            let counterValues: PrimitiveValue[] = dataView.metadata.columns.length > 2 && dataView.metadata.columns[2].roles['Counter'] ? dataView.categorical.values[1].values : null;
```
Comment (ignatvilesov) [inline src/visual.ts]: Is `counter` always at second index? It'd be better to find index of the counter by its role name in order to be more robust

File: src/visual.ts [hunk 8]
```diff
@@ -268,6 +268,11 @@ module powerbi.extensibility.visual {
                 categoricalValues: LineDotChartColumns<any[]> = LineDotChartColumns.getCategoricalValues(dataView),
                 settings: LineDotChartSettings = this.parseSettings(dataView);
 
+            if (counterValues && counterValues.length > 0) {
+                let fValue: any = counterValues[0];
+                settings.isCounterDateTime = fValue.getDate ? true : false;
```
Comment (ignatvilesov) [inline src/visual.ts]: That's ok. However, I'd prefer `fValue instanceof Date`

File: src/visual.ts [hunk 9]
```diff
@@ -461,29 +468,29 @@ module powerbi.extensibility.visual {
             this.main.attr('transform', SVGUtil.translate(this.layout.margin.left, this.layout.margin.top));
             this.legends.attr('transform', SVGUtil.translate(this.layout.margin.left, this.layout.margin.top));
             this.line.attr('transform', SVGUtil.translate(this.layout.margin.left + LineDotChart.LegendSize, 0));
-            this.axes.attr('transform', SVGUtil.translate(this.layout.margin.left + LineDotChart.LegendSize, 0));
-            this.axisX.attr('transform', SVGUtil.translate(0, this.layout.viewportIn.height - LineDotChart.LegendSize));
-            this.axisY2.attr('transform', SVGUtil.translate(this.layout.viewportIn.width - LineDotChart.LegendSize - LineDotChart.AxisSize, 0));
+                this.axes.attr('transform', SVGUtil.translate(this.layout.margin.left + LineDotChart.LegendSize, 0));
+                this.axisX.attr('transform', SVGUtil.translate(0, this.layout.viewportIn.height - LineDotChart.LegendSize));
+                this.axisY2.attr('transform', SVGUtil.translate(this.layout.viewportIn.width - LineDotChart.LegendSize - LineDotChart.AxisSize, 0));
```
Comment (ignatvilesov) [inline src/visual.ts]: Please fix the indent here.

File: src/visual.ts [hunk 10]
```diff
@@ -500,6 +507,23 @@ module powerbi.extensibility.visual {
...
             this.drawDots();
+            if (this.settings.axisoptions.show === true) {
+                this.SetAxisColor(this.settings.axisoptions.color);
+            } else {
+                this.SetAxisColor(LineDotChart.axesDefaultColor);
+            }
+        }
+
+        private SetAxisColor(color: string): void {
+            this.axisX.selectAll('line').style('stroke', function (d, i) { return color; });
```
Comment (ignatvilesov) [inline src/visual.ts]: Can we color inherit? If so, please ally these settings to this.axisX and this.axisY

File: capabilities.json (hunk 1 above)
Comment (vtkalek) [inline capabilities.json]: Because we need to have posibility to select date hierarchy

File: src/visual.ts (hunk 10 above)
Comment (vtkalek) [inline src/visual.ts]: They have different attributes to change colors? thats why i used separate defs

File: src/visual.ts [hunk 11]
```diff
@@ -483,25 +490,39 @@ module powerbi.extensibility.visual {
...
-
             linePathSelection
                 .exit().remove();
-
+            this.drawLine(linePathSelection);
+            this.drawClipPath(linePathSelection);
             this.drawDots();
         }
 
+        public SetAxisColor(color: string): void {
```
Comment (ignatvilesov) [inline src/visual.ts]: Name of the method must be in camel case. In other words, it should be `setAxisColor`

File: capabilities.json [hunk 12]
```diff
@@ -58,7 +75,7 @@
   ],
   "supportsHighlight": true,
   "objects": {
-    "lineoptions": {
+    "lineOptions": {
```
Comment (ignatvilesov) [inline capabilities.json]: You shouldn't rename properties. You should leave it as it was before.

File: capabilities.json [hunk 13]
```diff
@@ -82,10 +99,17 @@
         }
       }
     },
-    "dotoptions": {
+    "dotOptions": {
```
Comment (ignatvilesov) [inline capabilities.json]: You shouldn't rename properties. You should leave it as it was before.

File: capabilities.json [hunk 14]
```diff
@@ -113,7 +137,29 @@
...
+            "fill": {
+              "solid": {
+                "color": true
+              }
+            }
+          }
+        }
+      }
+    },
+    "counterOptions": {
```
Comment (ignatvilesov) [inline capabilities.json]: You shouldn't rename properties. You should leave it as it was before.

File: test/visualTest.ts [hunk 15]
```diff
@@ -155,6 +182,18 @@ namespace powerbi.extensibility.visual.test {
                     visualBuilder.dots.toArray().map($).forEach(e =>
                         colorHelper.assertColorsMatch(e.attr('fill'), color));
                 });
+                it("opacity", () => {
+                    let color: string = getRandomHexColor();
+                    debugger;
```
Comment (ignatvilesov) [inline test/visualTest.ts]: Please remove this statement.

**Pull request merged**